### DISCUSSION TITLE
Error When Trying to Delete Other People's Messages

### DISCUSSION THREAD

**Discussion by TsuniKu** [victornpb/undiscord#339] [1. Questions & Answers] [answered]
Just started using it now, whenever I would try to put in the User ID of the person's DM's I'm trying to delete, it goes through, did everything I need to do, then this error pops up whenever I press Start.

Error deleting message, API responded with status 403!	{"message":"Cannot execute action on a DM channel","code":50003}

No clue what I'm doing wrong, please help.

**Comment by victornpb** [comment, marked as answer]
**Discord doesn't allow you to delete the other person's messages on a DM chat.**

The only circumstance you are allowed to delete messages from others, is when you have the "Manage Messages" permission:
<img width="354" alt="image" src="https://user-images.githubusercontent.com/3372598/86542175-b0bc2880-bee9-11ea-98f0-f5b4093b0f11.png">

But it is HIGHLY recommended to use a bot in that case instead.
Some examples: [YAGPDB](https://yagpdb.xyz), [Dyno](https://dyno.gg), [Mee6](https://mee6.xyz)

**Comment by APratham** [comment]
Hey hi @victornpb, can MEE6 or Dyno delete messages posted by others on group DMs?

**Comment by SethFalco** [comment]
> Hey hi victornpb, can MEE6 or Dyno delete messages posted by others on group DMs?

I don't think that'd be possible.
1) You can't invite bots to group DMs as far as I know.
2) In DMs one can only delete their own messages.

**Comment by Fatih120** [comment]
Would you be willing to add a field that deletes a specific user's (by ID) messages? Bots are unable to delete messages older than 14 days and sometimes bots like Dyno stop responding with mass messages. For wanting to delete all of a bot's messages but not delete the channel as a result (as tasking as it sounds) would be amazing. This tool would be useful for such moderation purposes since the bots can't do the job for me.

**Comment by SethFalco** [comment]
> Would you be willing to add a field that deletes a specific user's (by ID) messages? Bots are unable to delete messages older than 14 days and sometimes bots like Dyno stop responding with mass messages. For wanting to delete all of a bot's messages but not delete the channel as a result (as tasking as it sounds) would be amazing. This tool would be useful for such moderation purposes since the bots can't do the job for me.

Sorry if I don't understand, but doesn't this already support that?
This can delete messages from other users in a guild, just not in a group or DM.
